openquake/hazardlib/gsim/sharma_2009.py

```diff
@@ -0,0 +1,241 @@
+
+    Reference:
+
+    Sharma, M. L., Douglas, J., Bungum, H., and Kotadia, J. (2009).
+    Ground-motion prediction equations based on data from the Himalayan and
+    Zagros regions. *Journal of Earthquake Engineering*, 13(8):1191–1210.
+    """
+
+    #: Supported tectonic region type is 'active shallow crust'
+    #: however as inndicated the introduction  the tectonics of the
+    #: Himalayas have a "great range of focal depths" (Sharma et
+    #: al., 2009, p. 1192).
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.ACTIVE_SHALLOW_CRUST
+
+    #: Set of :mod:`intensity measure types <openquake.hazardlib.imt>`
+    #: this GSIM can calculate. A set should contain classes from module
+    #: :mod:`openquake.hazardlib.imt`.
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([imt.SA])
+
+    #: Supported intensity measure component is the geometric mean of two
+    #: horizontal components
+    #: :attr:`~openquake.hazardlib.const.IMC.AVERAGE_HORIZONTAL`,
+    #: see p. 1200.
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.AVERAGE_HORIZONTAL
+
+    #: Only total standard deviation is supported, see Table 2, p. 1202.
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([const.StdDev.TOTAL])
+
+    #: Required site parameter Vs30 is used to set binary rock/soil
+    #: classification dummy variable, see equation (1) on p. 1200.
+    REQUIRES_SITES_PARAMETERS = set(('vs30',))
+
+    #: Required rupture parameters are magnitude and rake, see
+    #: equation (1) on p. 1200. Rake is used to distinguish between
+    #: reverse and strike-slip faulting, and to detect mis-application
+    #: of GMPE to normal faulting.
+    REQUIRES_RUPTURE_PARAMETERS = set(('rake', 'mag'))
+
+    #: Required distance measure is Joyner-Boore distance, see p. 1200
+    REQUIRES_DISTANCES = set(('rjb',))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, im_type, stddev_types):
+        # pylint: disable=too-many-arguments
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for specification of input and result values.
+        """
+
+        # extract dictionary of coefficients specific to required
+        # intensity measure type
+        coeffs = self.COEFFS[im_type]
+        coeffs.update(self.CONSTS)
+
+        log_mean = (self._compute_magnitude(rup, coeffs) +
+                    self._compute_distance(dists, coeffs) +
+                    self._get_site_amplification(sites, coeffs) +
+                    self._get_mechanism(rup, coeffs))
+        # Convert to g and thence to the natural logarithm
+        mean = log_mean*np.log(10.0) - np.log(g)
+
+        # Since sigma is not present in equation (1) it is not  entirely
+        # clear what it represents. We shall assume it requires the same
+        # processing that the mean value did.
+        log_stddevs = self._get_stddevs(coeffs, stddev_types, len(sites.vs30))
+        stddevs = log_stddevs*np.log(10.0)
```

Comment:
Can you put spaces around the * symbol (PEP8 convention)? Also I don't think the comments in lines 107 to 109 are particularly necessary.
Other than that the whole GMPE looks good to me (LGTM)